Migrations 3.0 (#858)
doctrine/migrations: UPGRADE.md

> # Upgrade to 3.0

Author: I personally prefer to keep it. But my opinion is not that strong on it, so if you prefer I can just drop it.

I can understand symfony that in some cases maintains 3 or more branches in parallel... thus the "upgrade" might be updated in the various branches.

> # Upgrade to 3.0

Reviewer: We'll leave it for now and do that in a separate PR.

# Upgrade to 3.0

> - The "version" is the FQCN of the migration class (exiting entries in the migrations table will be automatically updated).

Reviewer 2: ```suggestion
- The "version" is the FQCN of the migration class (existing entries in the migrations table will be automatically updated).
```

> - `MigrationsEventArgs` and `MigrationsVersionEventArgs` expose different API, please refer to the *Code BC breaks* section.

Reviewer 2: Please add a link, Github creates anchors for every title

## Console

> - Console output changed. The commands use different output style. If you were relying on specific output, 

Reviewer 2: ```suggestion
- Console output changed. The commands use a different output style. If you were relying on a specific output, 
```

  please update your scripts. 

>   Console output is not covered by the BC promise, so please try not to rely on specific output.

Reviewer 2: ```suggestion
  Console output is not covered by the BC promise, so please try not to rely on a specific output.
```

## Configuration files

*migrations.php Before*
```php
<?php

return [
    'name' => 'My Project Migrations',
    'migrations_namespace' => 'MyProject\Migrations',
    'table_name' => 'doctrine_migration_versions',
    'column_name' => 'version',
    'column_length' => 14,
    'executed_at_column_name' => 'executed_at',
    'migrations_directory' => '/data/doctrine/migrations-docs-example/lib/MyProject/Migrations',
    'all_or_nothing' => true,
    'check_database_platform' => true,
];
```
*migrations.php After*

```php
<?php

return [
    'name' => 'My Project Migrations',

    'table_storage' => [
        'table_name' => 'doctrine_migration_versions',
        'version_column_name' => 'version',
        'version_column_length' => 1024,
        'executed_at_column_name' => 'executed_at',
        'execution_time_column_name' => 'execution_time',
    ],

    'migrations_paths' => [
        'MyProject\Migrations' => '/data/doctrine/migrations/lib/MyProject/Migrations',
        'MyProject\Component\Migrations' => './Component/MyProject/Migrations',
    ],

    'all_or_nothing' => true,
    'check_database_platform' => true,
];
```

> In similar way also XML, YAML and JSON file changed. Please refer to the official documentation for more details.

Reviewer 2: ```suggestion
Files in XML, YAML or JSON also changed in a similar way. Please refer to the official documentation for more details.
```

## Code BC breaks

> Most of the code is protected by the `@internal` declaration and in a very rare case you should have dealt with the 

Reviewer 2: ```suggestion
Most of the code is protected by the `@internal` declaration and in a very rare cases you might have dealt with the 
```
